staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -286,7 +288,15 @@ func (rw *RetryWatcher) ResultChan() <-chan watch.Event {
 
 // Stop implements Interface.
 func (rw *RetryWatcher) Stop() {
```

Comment:
I checked when this method is actually called.
There might be further improvements. For example:

It doesn't make sense to [retry](https://github.com/kubernetes/kubernetes/blob/f638e2849fbddeb201b8290586a4747af99086bf/staging/src/k8s.io/client-go/tools/watch/retrywatcher.go#L195) the watcher past [this](https://github.com/kubernetes/kubernetes/blob/f638e2849fbddeb201b8290586a4747af99086bf/staging/src/k8s.io/client-go/tools/watch/retrywatcher.go#L141) point as it will be already closed, right ?

Reply:
@p0lyn0mial it's a bit confusing but `defer watcher.Stop()` is referring to the `watcher` returned from the `cache.Watcher` argument provided in `NewRetryWatcher`. If using a `client-go` implementation  (e.g. dynamic client) for `cache.Watcher`, it is already resilient to calling `Stop` multiple times. This `Stop` method on `RetryWatcher` acts as a layer above the actual watch stream to stop the `RetryWatcher` from retrying.